### ISSUE TITLE
Third fan not spinning for ROG Flow X16

### ISSUE THREAD

**Issue by cia19pl** [seerge/g-helper#104]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by PakNgaiJackyLee** [comment]
(Sorry I forgot to put in the description of the bug)

```
Describe the bug - 

Third, small fan not working on ASUS ROG Flow X16, causing the temperature to skyrocket during load.

To Reproduce - 

N/A

Expected behaviour - 

Third, small fan would ramp up as the load on the system increases.

Desktop - 

Windows 11 22H2
Chrome
V0.33

**Comment by seerge** [comment]
- How do you measure 3rd fan speed?
- What mode do you set (and do you apply custom fan curves)?

**Comment by seerge** [comment]
@cia19pl thanks. Currently app doesn't touch 3rd fan anyhow (means it doesn't disable or enable it). It could be that by applying a fan curve for other 2 fans ("apply" is checked on your screenshot) asus bios somehow decides to stop a 3rd (or may be not).

As a quick test - i would advice to uncheck "auto apply" (or just click factory defaults) and test if fan works or not.

May be in future I will add support for 3rd fan fan curves :)

**Comment by seerge** [comment]
[GHelper.zip](https://github.com/seerge/g-helper/files/11024015/GHelper.zip)
You can try this build, i have added support for the 3rd fan. But it appears only if bios responds with appropriate fan curve to the app.

**Comment by kafalier** [comment]
Maybe it'd be nice to add Sys Fan speed value somewhere on the main app interface in addition to CPU and GPU fan values for devices that have it. That way we can tell if it's working as intended or not. I think WMI for it should be `0x00110031`. Could do a PR myself but you'd probably do it faster.

**Comment by seerge** [comment]
[GHelper.zip](https://github.com/seerge/g-helper/files/11027246/GHelper.zip)
Build that should show mid fan speed on main screen

**Comment by kafalier** [comment]
Yes, mid fan speed shows. Fan editor also displays she settings for it as well, although since I don't use custom fan curves, I haven't tested whether they're working. But the read and the graph is there. Thanks.

**Comment by seerge** [comment]
Well, if chart shows - it means it can read fan curve from bios. If you don't click Apply (and don't check auto-apply) for fans, app will not apply any fan curve and bios will just use default ones.

**Comment by seerge** [comment]
Then I'm waiting for a response from @cia19pl

**Comment by seerge** [comment]
Great! I have pushed all changes into v0.35

**Comment by seerge** [comment]
Default fan curve is the one that your laptop's bios reports (app just reads and shows it) . I'm not a big fan of overwriting (by default) of what asus people have set :) But ofc you can always rise RPMs if that helps.
